manifests/profile/hathitrust/mounts.pp

```diff
@@ -34,44 +36,45 @@
     options => "size=${ramdisk_size}"
   }
 
-  package { 'nfs-common': }
-
-  $nfs_mount_options = {
-    ensure  => 'mounted',
-    fstype  => 'nfs',
-    require => ['Package[nfs-common]','File[/etc/resolv.conf]'],
-    tag     => 'private_network'
+  $smartconnect_mounts.each |$mount| {
+    nebula::monitored_nfs_mount { $mount:
+      smartconnect => true,
+      device       => "nas-macc.sc:/ifs${mount}"
+    }
   }
 
-  $mounts.each |$mount| {
+  $monitored_nfs_mounts.each |$mount| {
     file { $mount:
-      ensure => 'directory',
+      ensure => 'directory'
     }
+  }
+
+  create_resources(nebula::monitored_nfs_mount,$monitored_nfs_mounts)
 
-    mount { $mount:
-      name    => $mount,
-      device  => "nas-macc.sc:/ifs${mount}",
-      options => 'auto,hard',
-      *       => $nfs_mount_options
+  $other_nfs_mounts.each |$mount,$params| {
+    file { $mount:
+      ensure => 'directory'
     }
   }
 
+  create_resources(mount,$other_nfs_mounts,{
+    ensure  => 'mounted',
+    fstype  => 'nfs',
+    options => 'auto,hard'
```

Comment:
should still require & ensure `Package['nfs-common']` here

Reply:
should also consider having a parameter `monitored` on `monitored_nfs_mount` rather than repeating the default nfs mount options here